src/pkg/packager/validator/validator.go

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package validator contains functions for verifying zarf yaml files are valid
+package validator
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/config"
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+const (
+	validatorInvalidPrefix = "schema is invalid:"
+	validatorWarningPrefix = "zarf schema warning:"
+)
+
+// Validator holds the warnings/errors and messaging that we get from validation
+type Validator struct {
+	warnings           []string
+	errors             []error
+	jsonSchema         []byte
+	typedZarfPackage   types.ZarfPackage
+	untypedZarfPackage interface{}
+}
+
+func (v Validator) getFormatedError() error {
+	if !v.hasErrors() {
+		return nil
+	}
+	errorMessage := validatorInvalidPrefix
+	for _, errorStr := range v.errors {
+		errorMessage = fmt.Sprintf("%s\n - %s", errorMessage, errorStr.Error())
+	}
+	return errors.New(errorMessage)
+}
+
+func (v Validator) getFormatedWarning() string {
+	if !v.hasWarnings() {
+		return ""
+	}
+	return fmt.Sprintf("%s %s", validatorWarningPrefix, strings.Join(v.warnings, ", "))
+}
```

Comment:
@Racer159 What's the UX for this, I think a table would be more readable here than a really long string

Reply:
Yeah I was going to add newlines originally at least, but it looks like the current behavior of message.warn gets rid of the newlines and the string a paragraph. Maybe we use a different type of message or change that behavior?

Reply:
I ended up printing a table here, looks like this
![image](https://github.com/defenseunicorns/zarf/assets/37223396/977a3a5e-478a-4f90-be53-e0940152046a)